### DISCUSSION TITLE
Add option to collect user name during email registration

### DISCUSSION THREAD

**Discussion by bakua** [firebase/FirebaseUI-Flutter#69] [Ideas]
Hi there! I've been enjoying Firebase UI Auth, but I've noticed something. While social logins easily get display names, email sign-ups don't. Would it be possible to add an option to collect user names during email registration? It would really help keep a consistent experience for users. Thank you so much for considering this!

**Comment by lesnitsky** [comment]
This is fairly straightforward to implement. There's an [`EditableUserDisplayName`](https://pub.dev/documentation/firebase_ui_auth/latest/firebase_ui_auth/EditableUserDisplayName-class.html) widget. So you could handle `AuthStateChangeAction<UserCreated>` with a Navigation to your custom screen with `EditableUserDisplayName` on it.

Something along these lines:

```dart
SignInScreen(
  actions: [
    AuthStateChangeAction<UserCreated>((context, state) {
      final User(:displayName) = state.credential.user!;
      if (displayName != null) {
         // proceed to email verification or profile screen
      } else {
        Navigator.of(context).push(
          MaterialPageRoute(builder: (context) {
            return Scaffold(
              appBar: AppBar(
                title: const Text('What is your name?'),
              ),
              body: const Column(children: [
                EditableUserDisplayName(),
              ]),
            );
          }
        }),
      );
    }),
  ],
);
```

Feel free to submit a PR with your screen implementation, will be happy to review and assist with anything to help it land into the library. If you consider submitting a PR, keep in mind that the screen should be fairly generic and configurable.

**Reply by lesnitsky** [reply]
If anyone is willing to implement this, please leave a comment, so we have clarity that this is being worked on, to avoid duplicate work.

**Reply by YassinNouh21** [reply]
Hi
thanks for sharing I will start working on it !

**Reply by lesnitsky** [reply]
Tracked under #76
